File: csharp/all-your-base/AllYourBase.cs
using System;
using System.Linq;
using System.Collections.Generic;
using static System.Math;

public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        if(inputBase < 2 || outputBase < 2 || inputDigits.Any(n => n < 0 || n >= inputBase))
            throw new ArgumentException();

        var number = inputDigits.Reverse().Select((n, i) => 
            n * Pow(inputBase, i)).Sum();
        var result = new List<int>();

        while(number > 0)
        {
            var digit = number % outputBase;
            result.Insert(0, (int)digit);
            number = Math.Floor(number / outputBase);
        }

        return result.Count == 0 ? new[] { 0 } : result.ToArray();
    }
}

File: go/all-your-base/all_your_base.go
package allyourbase

import (
	"fmt"
	"math"
)

func ConvertToBase(inputBase int, inputDigits []int, outputBase int) ([]int, error) {
	result := []int{}
	if inputBase < 2 {
		return result, fmt.Errorf("input base must be >= 2")
	}
	if outputBase < 2 {
		return result, fmt.Errorf("output base must be >= 2")
	}
	base10 := 0
	for i, v := range inputDigits {
		if v >= inputBase || v < 0 {
			return result, fmt.Errorf("all digits must satisfy 0 <= d < input base")
		}
		base10 += v * int(math.Pow(float64(inputBase), float64(len(inputDigits)-(i+1))))
	}

	return convert(base10, outputBase, result), nil
}

func convert(remainder, outputBase int, result []int) []int {
	digit := remainder % outputBase
	result = append([]int{digit}, result...)
	quotient := int(math.Floor(float64(remainder) / float64(outputBase)))
	if quotient == 0 {
		return result
	}
	return convert(quotient, outputBase, result)
}
